src/main/java/bot/newsaggregator/service/TelegramBot.java

```diff
@@ -0,0 +1,113 @@
+import org.telegram.telegrambots.meta.api.objects.Message;
+import org.telegram.telegrambots.meta.api.objects.Update;
+import org.telegram.telegrambots.meta.api.objects.commands.BotCommand;
+import org.telegram.telegrambots.meta.api.objects.commands.scope.BotCommandScopeDefault;
+import org.telegram.telegrambots.meta.exceptions.TelegramApiException;
+
+import java.sql.Timestamp;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+
+@Slf4j
+@Component
+public class TelegramBot extends TelegramLongPollingBot {
+    @Autowired
+    private UserRepository userRepository;
+    private final BotConfig config;
+    static final String HELP_TEXT = "This bot is created for news aggregation.\n\n" +
+            "You can execute commands from main menu on the left or by typing a command:\n\n" +
+            "Type /start to see welcome message\n\n" +
+            "Type /my_data to see data stored about yourself\n\n" +
+            "Type /delete_data to delete stored user data\n\n" +
+            "Type /help to see message again\n\n" +
+            "Type /settings to users settings";
+
+    public TelegramBot(BotConfig config) {
+        this.config = config;
+        List<BotCommand> listOfCommands = new ArrayList<>();
+        listOfCommands.add(new BotCommand("/start", "get a welcome message"));
+        listOfCommands.add(new BotCommand("/my_data", "get your data stored"));
+        listOfCommands.add(new BotCommand("/delete_data", "delete my data"));
+        listOfCommands.add(new BotCommand("/help", "info how to use this bot"));
+        listOfCommands.add(new BotCommand("/settings", "set you preferences"));
+        try {
+            this.execute(new SetMyCommands(listOfCommands, new BotCommandScopeDefault(), null));
+        } catch (TelegramApiException e) {
+            log.error("Error setting bot's command list: " + e);
+        }
+    }
+
+    @Override
+    public String getBotUsername() {
+        return config.getBotName();
+    }
+
+    @Override
+    public String getBotToken() {
+        return config.getToken();
+    }
+
+    @Override
+    public void onUpdateReceived(Update update) {
+        if (update.hasMessage() && update.getMessage().hasText()) {
+            String messageText = update.getMessage().getText();
+            long chatId = update.getMessage().getChatId();
+            switch (messageText) {
+                case "/start":
+                    registerUser(update.getMessage());
+                    startCommandReceived(chatId, update.getMessage().getChat().getFirstName());
+                    break;
+                case "/help":
+                    sendMessage(chatId, HELP_TEXT);
+                    break;
+                default:
+                    sendMessage(chatId, "Sorry, command was not recognized");
+            }
+        }
+    }
+
+    private void registerUser(Message msg) {
+        if (userRepository.findById(msg.getChatId()).isEmpty()) {
```

Comment:
Make sure that race condition shouldn't happen. Consider wrapping the logic into transaction.